### PR TITLE
add strict mode to kafka change feeds and verify checkpoint management command

### PR DESCRIPTION
**Pull request by czue**
@dimagi/scale-team wanted to get this out and do some testing before fully plugging everything in.

this adds a way to check if a checkpoint offset is available in the kafka feed.

it also adds a strict mode to pillows so that they won't run if they have expired checkpoints (but doesn't turn it on yet), and a management command to run this that we can run on deploy after i've had a chance to test it a bit more.

wait for tests. probably easiest to review all at once.

### CODE REVIEW

File: corehq/apps/change_feed/consumer/feed.py [hunk 1]
```diff
@@ -3,11 +3,12 @@
 
 from django.conf import settings
 from kafka import KafkaConsumer
-from kafka.common import ConsumerTimeout
+from kafka.common import ConsumerTimeout, OffsetOutOfRangeError
```
Comment (dimagimon) [inline corehq/apps/change_feed/consumer/feed.py]: F401 'OffsetOutOfRangeError' imported but unused

File: corehq/apps/change_feed/consumer/feed.py [hunk 2]
```diff
@@ -3,11 +3,12 @@
...
 
 from corehq.apps.change_feed.data_sources import get_document_store
-from corehq.apps.change_feed.exceptions import UnknownDocumentStore
-from corehq.apps.change_feed.topics import get_multi_topic_offset, get_topic_offset
+from corehq.apps.change_feed.exceptions import UnknownDocumentStore, UnavailableKafkaOffset
```
Comment (dimagimon) [inline corehq/apps/change_feed/consumer/feed.py]: F401 'UnavailableKafkaOffset' imported but unused

File: corehq/apps/change_feed/consumer/feed.py [hunk 3]
```diff
@@ -3,11 +3,12 @@
...
+from corehq.apps.change_feed.topics import get_multi_topic_offset, get_topic_offset, \
```
Comment (dimagimon) [inline corehq/apps/change_feed/consumer/feed.py]: F401 'get_multi_topic_first_available_offsets' imported but unused

File: corehq/apps/change_feed/tests/test_kafka_change_feed.py [hunk 4]
```diff
@@ -1,11 +1,13 @@
 import uuid
 from django.test import SimpleTestCase, TestCase
 from kafka import KeyedProducer
-from kafka.common import KafkaUnavailableError
+from kafka.common import KafkaUnavailableError, OffsetOutOfRangeError
```
Comment (dimagimon) [inline corehq/apps/change_feed/tests/test_kafka_change_feed.py]: F401 'OffsetOutOfRangeError' imported but unused

**Comment by snopoke** [comment]
:+1: on :white_check_mark:

**Comment by czue** [comment]
:white_check_mark:

**Pull request merged**